### PR TITLE
releng: Add pull-release-sdk-integration-test job

### PR DESCRIPTION
**Pull request by xmudrii**
This PR adds the `pull-release-sdk-integration-test` job to the `release-sdk` repo which is used to run the integration tests.

This job uses the k8s-ci-bulider image because we require docker-in-docker (dind) for the integration tests.

This job is optional and supposed to be run manually using `/test pull-release-sdk-integration-test` (`always_run: false`) until we don't merge https://github.com/kubernetes-sigs/release-sdk/pull/31. After that PR is merged, I'll create a PR to enable this job for all PRs in the release-sdk repo.

/assign @saschagrunert @cpanato @puerco @palnabarun 
cc @kubernetes/release-engineering

### CODE REVIEW

**Review by cpanato — APPROVED** [review]
/lgtm
/hold for comments

File: config/jobs/kubernetes-sigs/release-sdk/release-sdk-presubmits.yaml
Comment (saschagrunert) [inline config/jobs/kubernetes-sigs/release-sdk/release-sdk-presubmits.yaml:20]: I would prefer using the preset label instead:
```yaml
    labels:
      preset-dind-enabled: "true"
```

Comment (xmudrii) [reply config/jobs/kubernetes-sigs/release-sdk/release-sdk-presubmits.yaml]: @saschagrunert Fixed, PTAL. I wasn't aware that we have this preset.

**Comment by xmudrii** [comment]
There are some failing tests when running tests with this image. I'll investigate, but in meanwhile, let's keep this on hold.
/hold

**Comment by xmudrii** [comment]
```
=== RUN   TestLastCommitSha
    git_test.go:595:
                Error Trace:    git_test.go:595
                Error:          Expected nil, but got: command /usr/bin/git commit --allow-empty -m Empty commit 1 did not succeed: Author identity unknown

                                *** Please tell me who you are.

                                Run

                                  git config --global user.email "you@example.com"
                                  git config --global user.name "Your Name"

                                to set your account's default identity.
                                Omit --global to set the identity only in this repository.

                                fatal: unable to auto-detect email address (got 'root@0255e836f914.(none)')
                Test:           TestLastCommitSha
```

The `k8s-ci-builder` image doesn't have git configured. The idea is to extend the Test target in the release-sdk repo to do that. I've created a PR for that: https://github.com/kubernetes-sigs/release-utils/pull/32

**Comment by xmudrii** [comment]
The issue mentioned above is fixed. The PR is updated to add a new `pull-release-sdk-integration-test` job instead of modifying the existing job. The new job is optional and supposed to be run manually until we don't merge https://github.com/kubernetes-sigs/release-sdk/pull/31. Therefore, merging this PR before https://github.com/kubernetes-sigs/release-sdk/pull/31 should be fine.

@saschagrunert @cpanato PTAL.

**Comment by puerco** [comment]
Label is changed @saschagrunert , let's roll.
/lgtm

**Pull request merged**
